packages/emptyset-reserve/contracts/reserve/ReserveBase.sol

```diff
@@ -0,0 +1,181 @@
+    /// @return redemptionAmount The quantity of underlying assets redeemed
+    function _redeem(UFixed18 amount) internal returns (UFixed18 redemptionAmount) {
+        redemptionAmount = amount.mul(redeemPrice());
+
+        IDSU(Token18.unwrap(dsu)).burn(UFixed18.unwrap(amount));
+        emit Redeem(msg.sender, amount, redemptionAmount);
+    }
+
+    /// @notice Issues new DSU
+    /// @dev Internal helper function
+    /// @param amount The quantity of DSU to issue
+    function _issue(UFixed18 amount) internal {
+        IDSU(Token18.unwrap(dsu)).mint(UFixed18.unwrap(amount));
+        emit Issue(msg.sender, amount);
+    }
+
+    /// @notice Computes the reserve's target allocation in the underlying strategy after removing `amount` assets
+    /// @param amount The quantity of assets to remove from the reserve
+    /// @return allocated The quantity of assets allocated to the underlying strategy
+    /// @return target The target quantity of assets to allocate to the underlying strategy
+    function _compute(UFixed18 amount) private view returns (UFixed18 allocated, UFixed18 target) {
+        UFixed18 unallocated = _unallocated();
+        allocated = _allocated();
+        target = unallocated.add(allocated).sub(amount).mul(allocation);
+    }
+
+    /// @notice Updates the reserve's allocation in the underlying strategy from `allocated` to `target`
+    /// @param amount The quantity of assets to remove from the reserve
+    function _allocate(UFixed18 amount) private {
+        (UFixed18 allocated, UFixed18 target) = _compute(amount);
+        _update(allocated, target);
+    }
+
+    /// @notice Returns the quantity of assets allocated to the reserve's underylying strategy (ex. USD-value of cUSDC)
+    /// @return The quantity of assets allocated to the reserve's underylying strategy
+    function _unallocated() internal virtual view returns (UFixed18);
+
+    /// @dev Returns the quantity of assets unallocated in the reserve (ex. USDC)
+    /// @return The quantity of assets unallocated in the reserve
+    function _allocated() internal virtual view returns (UFixed18);
+
+    /// @notice Pulls underlying assets from the caller
+    /// @param amount The quantity of underlying assets to pull (scaled to 18 decimal places)
+    function _pull(UFixed18 amount) internal virtual;
+
+    /// @dev Pushes underlying assets to the caller
+    /// @param amount The quantity of underlying assets to push (scaled to 18 decimal places)
+    function _push(UFixed18 amount) internal virtual;
+
+    /// @notice Updates the reserve's allocation in the underlying strategy from `collateral` to `target`
+    /// @param allocated The quantity of assets allocated to the underlying strategy
+    /// @param target The target quantity of assets to allocate to the underlying strategy
+    function _update(UFixed18 allocated, UFixed18 target) internal virtual;
+
+    /// @dev Only the coordinator can call the function
+    modifier onlyCoordinator {
```

Comment:
I did this on purpose to keep it simpler, because the owner could always create a proposal that "took back" coordinator rights, executed the action, then gave back rights to coordinator atomically.

Reply:
That would take a full governance proposal process though right? Might not be fast enough for this case